plugins/modules/purefa_pod_replica.py

```diff
@@ -85,133 +92,165 @@
-        for link in range(0, len(rlinks)):
-            if rlinks[link]["local_pod_name"] == module.params["name"]:
-                return rlinks[link]
-        return None
-    except Exception:
+    api_version = array.get_rest_version()
+    if LooseVersion(CONTEXT_VERSION) <= LooseVersion(api_version):
+        res = array.get_pod_replica_links(
+            local_pod_names=[module.params["name"]],
+            context_names=[module.params["context"]],
+            total_item_count=True,
+        )
+    else:
+        res = array.get_pod_replica_links(
+            local_pod_names=[module.params["name"]], total_item_count=True
+        )
+    if res.total_item_count == 0:
         return None
-
-
-def _get_arrays(array):
-    """Get Connected Arrays"""
-    arrays = []
-    array_details = array.list_array_connections()
-    for arraycnt in range(0, len(array_details)):
-        arrays.append(array_details[arraycnt]["array_name"])
-    return arrays
+    else:
+        return list(res.items)[0]
 
 
 def update_rl(module, array, local_rl):
     """Create Pod Replica Link"""
+    api_version = array.get_rest_version()
     changed = False
     if module.params["pause"] is not None:
-        if local_rl["status"] != "paused" and module.params["pause"]:
+        if local_rl.status != "paused" and module.params["pause"]:
             changed = True
             if not module.check_mode:
-                try:
-                    array.pause_pod_replica_link(
+                if LooseVersion(CONTEXT_VERSION) <= LooseVersion(api_version):
+                    res = array.pause_pod_replica_link(
+                        local_pod_name=module.params["name"],
+                        context_names=[module.params["context"]],
+                        remote_pod_name=local_rl["remote_pod_name"],
+                    )
+                else:
+                    res = array.pause_pod_replica_link(
                         local_pod_name=module.params["name"],
                         remote_pod_name=local_rl["remote_pod_name"],
                     )
-                except Exception:
+                if res.status_code != 200:
                     module.fail_json(
-                        msg="Failed to pause replica link {0}.".format(
-                            module.params["name"]
+                        msg="Failed to pause replica link {0}. Error: {1}".format(
+                            module.params["name"], res.errors[0].message
                         )
                     )
-        elif local_rl["status"] == "paused" and not module.params["pause"]:
+        elif local_rl.status == "paused" and not module.params["pause"]:
             changed = True
             if not module.check_mode:
-                try:
-                    array.resume_pod_replica_link(
+                if LooseVersion(CONTEXT_VERSION) <= LooseVersion(api_version):
+                    res = array.resume_pod_replica_link(
```

Comment:
array doesn't have resume_pod_replica_link attribute, should use "patch_pod_replica_links"